src/sentry/incidents/logic.py

```diff
@@ -591,6 +600,146 @@ class AlertRuleNameAlreadyUsedError(Exception):
+                            False,
+                            "CRITICAL trigger must have an alert threshold below WARNING trigger",
+                        )
+                    elif triggers[0]["resolveThreshold"] < triggers[1]["resolveThreshold"]:
+                        return (
+                            False,
+                            "CRITICAL trigger must have a resolution threshold above (or equal to) WARNING trigger",
+                        )
+        else:
+            return (
+                False,
+                "Must send 1 or 2 triggers - either 1 CRITICAL trigger, or 1 CRITICAL and 1 WARNING trigger",
+            )
+
+        # Triggers have passed checks. Check that all triggers have at least one action now.
+        for trigger in triggers:
+            actions = trigger.get("actions", [])
+            if actions == []:
+                return False, trigger["label"] + " trigger must have an action."
+    else:
+        return False, "Must include at least one trigger"
+
+    return True, "Valid"
+
+
+def create_alert_rule_unified(data, organization, access):
+    """
+    Creates an alert rule for an organization, just like create_alert_rule, but
+    this function also accepts trigger and action data, and creates those as well.
+    This function uses the serializers which end up calling their underlying create functions - such as `create_alert_rule`.
+
+    Additionally, it enforces these criteria:
+        -1 Trigger must be present of CRITICAL_TRIGGER type.
+        -A second trigger can be supplied, of a WARNING_TRIGGER type.
+            - If the optional warning trigger is supplied, the thresholdType must be the same (???)
+        -If two triggers are supplied, their alert and resolution thresholds must:
+            - Critical alert thresholds must be triggered *before* the warning (whether this means the threshold has to be larger or smaller depends on the thresholdType)
+            - Critical resolution thresholds must be triggered before or at the same time as the warning
+
+        -Each trigger is enforced to have at least one action.
+
+    If those enforced criteria are not met, nothing is created and this function will return an error status and message.
+    If it is created, it will return the created rule as the value of "rule" in the returned dict.
+
+    :return:
+        {
+            "error": True/False, # required.
+            "rule": The created `AlertRule`, # if 'error' is False, this is required.
+            "message": A unicode string representing the error encountered. Optional if 'error' is False, required if 'error' is True
+        }
+    """
+
+    is_valid, message = validate_unified_alert_rule(data)
+    if not is_valid:
+        return {"error": True, "message": message}
+
+    from sentry.incidents.endpoints.serializers import (
+        AlertRuleSerializer,
+        AlertRuleTriggerSerializer,
+        AlertRuleTriggerActionSerializer,
+    )
```

Comment:
Why does this import need to be deferred?

Reply:
There seems to be some sort of circular import dependency error?